packages/flutter/test/widget/reparent_state_harder_test.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/src/widgets/basic.dart';
+import 'package:flutter/src/widgets/framework.dart';
+import 'package:flutter/src/widgets/layout_builder.dart';
+import 'package:flutter_test/flutter_test.dart' hide TypeMatcher;
+
+class OrderSwitcher extends StatefulWidget {
+  OrderSwitcher({ Key key, this.a, this.b }) : super(key: key);
+
+  final Widget a;
+  final Widget b;
+
+  @override
+  OrderSwitcherState createState() => new OrderSwitcherState();
+}
+
+class OrderSwitcherState extends State<OrderSwitcher> {
+
+  bool _aFirst = true;
+
+  void switchChildren() {
+    setState(() {
+      _aFirst = false;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> children = <Widget>[];
+    if (_aFirst) {
+      children.add(new KeyedSubtree(child: config.a));
+      children.add(config.b);
+    } else {
+      children.add(new KeyedSubtree(child: config.b));
+      children.add(config.a);
+    }
+    return new Stack(
+      children: children
+    );
+  }
+}
+
+class DummyStatefulWidget extends StatefulWidget {
+  DummyStatefulWidget(Key key) : super(key: key);
+
+  @override
+  DummyStatefulWidgetState createState() => new DummyStatefulWidgetState();
+}
+
+class DummyStatefulWidgetState extends State<DummyStatefulWidget> {
+  @override
+  Widget build(BuildContext context) => new Text('LEAF');
+}
+
+class RekeyableDummyStatefulWidgetWrapper extends StatefulWidget {
+  RekeyableDummyStatefulWidgetWrapper({ this.child, this.initialKey });
+  final Widget child;
+  final GlobalKey initialKey;
+  @override
+  RekeyableDummyStatefulWidgetWrapperState createState() => new RekeyableDummyStatefulWidgetWrapperState();
+}
+
+class RekeyableDummyStatefulWidgetWrapperState extends State<RekeyableDummyStatefulWidgetWrapper> {
+  GlobalKey _key;
+
+  @override
+  void initState() {
+    super.initState();
+    _key = config.initialKey;
+  }
+
+  void _setChild(GlobalKey value) {
+    setState(() {
+      _key = value;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new DummyStatefulWidget(_key);
+  }
+}
+
+void main() {
+  testWidgets('Flutter Gallery app smoke test', (WidgetTester tester) async {
```

Comment:
> I don't think this is string you want :-).

Oh man, I missed the last remaining evidence that this came from the gallery smoketest!

> I think it would be useful to include a comment about this being a regression test for https://github.com/flutter/flutter/issues/5588

Added a comment.

> A brief summary of what's going on here would be helpful too. This test's setup is more complex than most.

But then it'll be obvious I don't really know what's going on... :-P